pkg/apis/config/types.go

```diff
@@ -45,6 +47,78 @@ type ETCD struct {
 	Backup ETCDBackup
 }
 
+type CSI struct {
+	// CSIAttacher is the configuration for the csi-attacher
+	CSIAttacher *CSIAttacher
+	// CSIDriverCinder is the configuration for the csi-driver-cinder
+	CSIDriverCinder *CSIDriverCinder
+	// CSIProvisioner is the configuration for the csi-provisioner
+	CSIProvisioner *CSIProvisioner
+	// CSIResizer is the configuration for the csi-resizer
+	CSIResizer *CSIResizer
+	// CSISnapshotController is the configuration for the csi-shapshot-controller
+	CSISnapshotController *CSISnapshotController
+	// CSISnapshotter is the configuration for the csi-snapshotter
+	CSISnapshotter *CSISnapshotter
+	// CSILivenessProbe is the configuration for the csi-liveness-probe
+	CSILivenessProbe *CSILivenessProbe
+}
+
+type CSIAttacher struct {
+	// RetryIntervalStart The exponential backoff for failures.
+	RetryIntervalStart *string
+	// RetryIntervalMax The exponential backoff maximum value.
+	RetryIntervalMax *string
+	// ReconcileSync Resync frequency of the attached volumes with the driver.
+	ReconcileSync *string
+	// Timeout Timeout of all calls to the container storage interface driver.
+	Timeout *string
+	// Verbose The verbosity level.
+	Verbose *string
+}
+
+type CSIDriverCinder struct {
```

Comment:
Many of those structs are exactly the same. How about embed the things that they have in common e.g. `Verbose` or `Timeout`?

Reply:
Yeah I didn't really like the solution either tbh. I guess we could try and identify common args but these could also change. And the fact that we have like 6-7 different csi components doesn't make it necessarily easier. Would be cool if we could have some super generic option where u just enter what you want like `extraArgs` helm-chart kind off stuff.

Reply:
Is that something you would be interested in @dkistner ?

Reply:
Well at the end I guess we need something like this:

```go
type CSI struct {
	// CSIAttacher is the configuration for the csi-attacher
	CSIAttacher *CSIAttacher
	// CSIDriverCinder is the configuration for the csi-driver-cinder
	CSIDriverCinder *CSIBaseArgs
	// CSIProvisioner is the configuration for the csi-provisioner
	CSIProvisioner *CSIBaseArgs
	// CSIResizer is the configuration for the csi-resizer
	CSIResizer *CSIBaseArgs
	// CSISnapshotController is the configuration for the csi-shapshot-controller
	CSISnapshotController *CSIBaseArgs
	// CSISnapshotter is the configuration for the csi-snapshotter
	CSISnapshotter *CSIBaseArgs
	// CSILivenessProbe is the configuration for the csi-liveness-probe
	CSILivenessProbe *CSIBaseArgs
}

type CSIAttacher struct {
        CSIBaseArgs
	// RetryIntervalStart The exponential backoff for failures.
	RetryIntervalStart *string
	// RetryIntervalMax The exponential backoff maximum value.
	RetryIntervalMax *string
	// ReconcileSync Resync frequency of the attached volumes with the driver.
	ReconcileSync *string
}

type CSIBaseArgs struct {
	// Timeout Timeout of all calls to the container storage interface driver.
	Timeout *string
	// Verbose The verbosity level.
	Verbose *string
}
```

I guess with something like this we can avoid a lot DRY also for the helm values generation in the controlplane controller.